chronos/apps/http/httpclient.nim

```diff
@@ -604,18 +666,35 @@ proc acquireConnection(
   else:
     let conn =
       block:
-        let conns = session.connections.getOrDefault(ha.id)
-        if len(conns) > 0:
-          var res: HttpClientConnectionRef = nil
-          for item in conns:
-            if item.state == HttpClientConnectionState.Ready:
-              res = item
-              break
-          res
-        else:
-          nil
+        var idleConnections: seq[HttpClientConnectionRef]
+        let
+          timestamp = Moment.now()
+          res =
+            block:
+              var cres: HttpClientConnectionRef = nil
+              session.connections.withValue(ha.id, connections):
+                for connection in connections[]:
+                  if connection.isReady():
+                    if connection.isIdle(timestamp, session.idleTimeout):
```

Comment:
shouldn't this happen on a timer? ie if nobody is requesting new connections, they'll stay forever?